web/src/engine/websites/Toonkor.ts

```diff
@@ -0,0 +1,43 @@
+﻿import { Tags } from '../Tags';
+import icon from './Toonkor.webp';
+import { DecoratableMangaScraper } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import * as CoreView from './decorators/CoreView';
+
+function ChapterExtractor(element: HTMLElement) {
+    return {
+        id: element.dataset.role,
+        title: element.textContent.trim()
+    };
+}
+
+const pageScript = `
+    new Promise (resolve => {
+        const dom =  new DOMParser().parseFromString(toon_img, 'text/html');
+        resolve([...dom.querySelectorAll('img')].map(img => new URL(img.src, window.location.origin).href));
+    })
+`;
+
+@Common.MangaCSS(/https:\/\/toonkor\d+\.com\/[^/]+$/, 'table.bt_view1 td.bt_title')
+@CoreView.MangasSinglePageCSS(['/웹툰/연재?fil=제목', '/웹툰/완결?fil=제목'], 'div.section-item-title a#title', Common.AnchorInfoExtractor())
+@Common.ChaptersSinglePageCSS('td.content__title', ChapterExtractor)
+@Common.PagesSinglePageJS(pageScript, 1500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('toonkor', `Toonkor`, 'https://toonkor463.com', Tags.Media.Manhwa, Tags.Language.Korean, Tags.Source.Aggregator, Tags.Accessibility.DomainRotation);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async Initialize(): Promise<void> {
+        //this.URI.href = (await FetchCSS<HTMLAnchorElement>(new Request('https://t.me/s/new_toonkor'), 'a[href^="https://toonkor"]')).at(-1)?.origin ?? this.URI.href;
```

Comment:
The telegram channel seems to stop working for a while now, so the comment may be removed